### ISSUE TITLE
Default point of rotate_* filters

### ISSUE THREAD

**Issue by tkoyama010** [pyvista/pyvista#1579]
Some filters default to the mesh center, and that would actually make sense here. Too bad the rotate_* filters don't do that already.

_Originally posted by @adeak in https://github.com/pyvista/pyvista/pull/1569#discussion_r684923956_

**Comment by adeak** [comment]
Thanks for opening this. However, I'm a bit worried about changing the default here. This would mean that code that used to work still works, but suddenly and silently gives different results, almost always (since, statistically speaking, most meshes aren't centered on the origin).

**Comment by tkoyama010** [comment]
@adeak Thanks for comment. It is a very good point. We should be cautious about the impact on users. What do you think about creating a new method with a slightly different name? (And Deprecate Warning in old method.)

**Comment by adeak** [comment]
Honestly, I have no idea what the best strategy is. Options I can think of:

1. do nothing
2. just change the default
3. add a loud warning telling users that the behaviour will change in `n` (`n=2`?) versions, and they should be explicit about the center in the meantime
4. deprecate the old function and add a new one, as you suggested.

If we go with 1. there's also a side-concern:
    1.b. do we prefer to have the new flip_* methods have sane default, or do we prefer to have the new flip_* methods match the defaults of the rotate_* methods? We can't have both.
    
@akaszynski thoughts?

**Comment by akaszynski** [comment]
Good discussion.

Do nothing with existing methods.

We then add new methods to allow rotation about center (I think `rotate_x_about_center` sounds good).  Same with the flip methods.

Thoughts?

**Comment by tkoyama010** [comment]
@akaszynski Thanks for comment. +1 for adding `rotate_x_about_center` .

**Comment by MatthewFlamm** [comment]
There are already a lot of methods for datasets.  It is always growing too.  I'm not enthusiastic about adding more functions to get around a default parameter issue.  But, the problem with option 3 is that there is no way to future-proof your code without unneeded code logic, i.e. everyone would have to specify the point they want until the deprecation period is over and there is no longer a point in having a default.   So I reluctantly also vote for option 1 or 1a for now.

The only thing I would consider for option 3 is to write down the preferred default behaviors and then identify all the inconsistent methods. The we could have the breaking changes in the first major release, i.e. v1.0.0.  This is a lot of effort required to implement however.

**Comment by akaszynski** [comment]
> There are already a lot of methods for datasets

That's quite true, and if we had two rotate methods we'd be polluting our namespace a bit as well.  In that sense, we could then change the default behavior of this method for a `v1.0.0` release and add "pending deprecation".  `point=None` for default, which defaults to the `dataset.center`.

**Comment by adeak** [comment]
I agree with Matthew that adding new functions would lead to too much bloat. I guess making v1.0 a larger breaking change would be a way out. But we should then tend to the planned deprecation using a dedicated (and kept up-to-date) tracking issue, and try to communicate the planned future breaking changes in a very visible way.

**Comment by tkoyama010** [comment]
Thinking about versioning is a good point. I think "[Semantic Versioning](https://semver.org)" is a good document to use.

Given a version number MAJOR.MINOR.PATCH, increment the:
1. MAJOR version when you make incompatible API changes,
1. MINOR version when you add functionality in a backwards compatible manner, and
1. PATCH version when you make backwards compatible bug fixes.

Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
Can I add this versioning Policy to [Branching Model](https://github.com/pyvista/pyvista/blob/main/CONTRIBUTING.md#branching-model) ? Then I can solve this problem.

**Comment by adeak** [comment]
Doing proper semver needs a lot of commitment, and it's very far from our current "version on a whim" model :P Some more order would be nice in the long run, but we shouldn't add too much overhead to the reviewing process like that.

Even without semver v1.0 generally means a first public stable version. That's why breakages _might_ be more OK. But see also the story of Python 2 for too much breakage.

**Comment by akaszynski** [comment]
We mostly follow semver:
- Our patch releases follow semver perfectly.
- Minor sorta follows this.  We do make backwards incompatible changes, in defense we're not v1 and our API isn't 100% grounded and we should be more fluid given that we're all learning.
- Our v1 should "freeze" our API, rather than cause a discrete API change (i.e. Python 2 vs 3).  Should we ever make it to v2, we can do that.
